### PR TITLE
tronfig checks valid nodes

### PR DESCRIPTION
**Pull request by chlgit**
The command is something like:
   tronfig tron-playground.yaml -V -n robj-test-service -m ../tron/playground/MASTER.yaml

The output is something like:
  Unknown node name batch1-uswest1adevcfff at robj-test-service.NamedConfigFragment.jobs.Job.mesos-test.actions.Action.sleep.node

### CODE REVIEW

**Comment by qui** [comment]
I don't think we should have to add validation code for this, Tron will already check that the nodes are valid if you use `validate_config_mapping`. 

Also the options are getting very convoluted here... I wish we had separate subcommands for validate, print, etc instead of making them all optional arguments, but maybe it's too late to change the interface now?

**Comment by chlgit** [comment]
Separated subcommands is a good idea, but not sure how much time to refactor. Also, I imagine Tronfig is rarely used manually (it's used for deploy or pre-commit hook for most of the use cases), so it's okay to be a little convoluted?

File: tests/config/config_parse_test.py [hunk 1]
```diff
@@ -455,6 +455,128 @@ def test_attributes(self):
...
+
+    def test_invalid_job_node_with_master_context(self):
+        master_context = dict(
+            nodes=[
+                dict(
+                    name="node0",
+                    hostname="node0",
+                )
+            ],
+            node_pools=[
```
Comment (qui) [inline tests/config/config_parse_test.py]: nit: let's take these lines out because they don't affect the test.

File: tron/config/config_parse.py [hunk 2]
```diff
@@ -716,12 +716,18 @@ def post_validation(self, config, config_context):
 valid_named_config = ValidateNamedConfig()
 
 
-def validate_fragment(name, fragment):
+def validate_fragment(name, fragment, master_context=None):
```
Comment (qui) [inline tron/config/config_parse.py]: Can we call this master_config everywhere, since it's not a Context object?

**Pull request merged**